packaging/helm-charts/helm3/strimzi-kafka-operator/templates/020-RoleBinding-strimzi-cluster-operator.yaml

```diff
@@ -1,6 +1,10 @@
 {{- if .Values.rbac.create -}}
 {{- $root := . -}}
-{{- range append .Values.watchNamespaces .Release.Namespace }}
+{{- $watchNamespaces := .Values.watchNamespaces -}}
+{{- if $root.Values.watchAnyNamespace }}
+  {{- $watchNamespaces = list -}}  
```

Comment:
If "watchAnyNamespace" is set to true, we will set variable watchNamespaces to empty List, and  ignore the values of ".Values.watchNamespaces"

Loop over new variable($watchNamespaces) instead of ".Values.watchNamespaces"